### ISSUE TITLE
Display Screen Crash in Space Astronomy modpack

### ISSUE THREAD

**Issue by MJRLegends** [micdoodle8/Galacticraft#2891]
Sorry im currently away so im unable to fully look at this myself and currently trying to get stuff done from my phone.

See https://github.com/MJRLegends/Space-Astronomy-2-Feedback-/issues/44

**Comment by radfast** [comment]
I've responded on the Space Astronomy issues thread.  It seems to be a dimension registration issue between client and server, the MapUtil code is being called from a ```null``` world which should be impossible.

We can add a null check, but I'd like to understand more about how this situation is arising.

>
    java.lang.NullPointerException: Exception in server tick loop
        at micdoodle8.mods.galacticraft.core.util.MapUtil.sendOrCreateMap(MapUtil.java:241)
        at micdoodle8.mods.galacticraft.core.network.PacketSimple.handleServerSide(PacketSimple.java:1212)
        at micdoodle8.mods.galacticraft.core.network.GalacticraftPacketHandler.tick(GalacticraftPacketHandler.java:68)

**Comment by MJRLegends** [comment]
Yeah, i will try to test it myself when im able to get on my computer in a few days time. Also the mod pack is Space Astronomy 2 which is a 1.10.2 mc mod pack and can be found here https://minecraft.curseforge.com/projects/space-astronomy-2
I know its a similar name so just wanted to make sure it wasnt confused with my Space Astronomy 1 mod pack which is mc 1.7.10

**Comment by MJRLegends** [comment]
Also thought its important to mention it contains my Galacticraft addon ExtraPlanets

**Comment by radfast** [comment]
+ about 100 other mods, @MJRLegends 

Galacticraft alone (+ a handful of other popular mods) does not have this problem.  Please test your pack and figure out - maybe using the binary search method - which mod combined with GC is causing this.

Whatever is causing this is breaking, at a fairly fundamental level, calls like getting the ```WorldProvider``` for dimension 0 and then getting the world for that ```WorldProvider```.  This is likely to mess up multiple things in Galacticraft, not only the Display Screens.

**Comment by MJRLegends** [comment]
I have a feeling its just a one user issue as ive had no issues myself and no one else has reported it

**Comment by MJRLegends** [comment]
Ok @radfast It seems its Astral Sorcery and Galacticraft together that is causing the issue, i have informed the dev of Astral Sorcery :P

**Comment by MJRLegends** [comment]
There is a a temp fix of the following below and the full fix will be done by the Astral Sorcery dev soon

```
disable the full AS skyrender by adding dimension 0 to the weak sky renders.
then you lose AS' skyrendering and eclipses and stuff, but it won't crash anymore.
```

**Comment by HellFirePvP** [comment]
AS 1.6.1 fixes the issue by re-initializing the WorldProvider. fixed

**Comment by MJRLegends** [comment]
Is this available for mc 1.10.2

**Comment by MJRLegends** [comment]
ping @HellFirePvP

**Comment by radfast** [comment]
Closing as it seems like this was on Astral Sorcery's side by doing something non-standard to the WorldProvider ...

**Comment by HellFirePvP** [comment]
Fixed in the 1.12 release, I currently don't have the spare time to backport most of the things to 1.11.2/1.10.2 to fix this properly though. Might do so in the future.
